### PR TITLE
Add more tests for super user domain

### PR DESCRIPTION
**Pull request by mgao0**
This commit add one more test for super user case for X509ZnodeGroupAclProvider. There are two cases where a user is granted as super user: (1) User URI matches with the super user URI (2) User URI is mapped to a super user domain. The previous test only covers (2), this commits add test for (1).

### CODE REVIEW

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclProperties.java [hunk 1]
```diff
@@ -115,4 +119,10 @@ public static ZNodeGroupAclProperties getInstance() {
     }
     return superUserDomainNames;
   }
+
+  public static void reset() {
```
Comment (narendly) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclProperties.java]: Why is this needed?
Maybe call this `clearProperties()` and put VisibleForTesting?

**Review by narendly — APPROVED** [review]
Mostly good, let's address comments?

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclProperties.java (hunk 1 above)
Comment (mgao0) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclProperties.java]: This is needed because ZNodeGroupAclProperties is only instantiated once, but sometimes in test I want to use different properties. Updated the code.
